src/screens/login/LoginScreen.js

```diff
@@ -147,7 +149,10 @@ const LoginSteps = ({
       try {
         const client = await createClient(instance)
 
-        const { kdfIterations, name } = await fetchPublicData(client)
+        const { kdfIterations, name, locale } = await fetchPublicData(client)
+
+        // If we are able to get a locale from the public data, we update the language asap
+        if (locale) void changeLanguage(locale)
```

Comment:
`setInstanceData` is called only for classic login, in case of magicLink based auth then `startMagicLinkOAuth` is called and in case of OIDC oauth then `startOidcOAuth` is called. During those scenario we can have the 2FA screen displayed that should be translated too.

Also I'm wondering if we have enough data to do the same on the OnboardingScreen?

Reply:
I suggest doing this in a following PR as getting locale info before full login wasn't really in the scope of this feature at the start

Reply:
I'm ok with that 👍